libindi/libs/indicom.c

```diff
@@ -435,26 +433,47 @@ int tty_read(int fd, char *buf, int nbytes, int timeout, int *nbytes_read)
     if (tty_debug)
         IDLog("%s: Request to read %d bytes with %d timeout for fd %d\n", __FUNCTION__, nbytes, timeout, fd);
 
-    while (nbytes > 0)
+    if (ttyGeminiUdpFormat)
+    {
+        numBytesToRead = nbytes + 8;
+    }
+
+    while (numBytesToRead > 0)
     {
         if ((err = tty_timeout(fd, timeout)))
             return err;
 
-        bytesRead = read(fd, buf + (*nbytes_read), ((uint32_t)nbytes));
+        bytesRead = read(fd, readBuffer + (*nbytes_read), ((uint32_t)numBytesToRead));
 
         if (bytesRead < 0)
             return TTY_READ_ERROR;
 
         if (tty_debug)
         {
-            IDLog("%d bytes read and %d bytes remaining...\n", bytesRead, nbytes - bytesRead);
+            IDLog("%d bytes read and %d bytes remaining...\n", bytesRead, numBytesToRead - bytesRead);
             int i = 0;
             for (i = *nbytes_read; i < (*nbytes_read + bytesRead); i++)
                 IDLog("%s: buffer[%d]=%#X (%c)\n", __FUNCTION__, i, (unsigned char)buf[i], buf[i]);
         }
 
         *nbytes_read += bytesRead;
-        nbytes -= bytesRead;
+        numBytesToRead -= bytesRead;
+    }
+
+    if (ttyGeminiUdpFormat)
+    {
+        if (*(int *)(readBuffer) != sequenceNumber)
+        {
+            // Not the right reply just do the read again.
+            return tty_read(fd, buf, nbytes, timeout, nbytes_read);
+        }
+
+        *nbytes_read -= 8;
+        strncpy(buf, readBuffer+8, *nbytes_read);
+    }
+    else
+    {
+        strncpy(buf, readBuffer, *nbytes_read);
```

Comment:
Strncpy here is a no-no too, same thing as above with writes

Reply:
Sorry I just saw this, was this addressed?

Reply:
Yes those issues were fixed in another PR. I don't remember which one though.